src/metabase/models/humanization.clj

```diff
@@ -82,10 +83,15 @@
   :visibility :settings-manager
   :audit      :raw-value
   :getter     (fn []
-                (let [strategy (setting/get-value-of-type :keyword :humanization-strategy)]
-                  ;; actual advanced method has been excised. Use `:simple` instead if someone had specified
-                  ;; `:advanced`.
-                  (if (= strategy :advanced)
-                    :simple
-                    strategy)))
+                (let [strategy (setting/get-value-of-type :keyword :humanization-strategy)
+                      valid-values (set (keys (methods u.humanization/name->human-readable-name)))
```

Comment:
How about we exclude `:advanced` from `valid-values` as well? It hasn't been supported for a long time and the implementation of `name->human-readable-name` just defaults to `:simple`. So if someone is trying to set `:advanced` they should probably also see an error.

Reply:
While `:advanced` hasn't been used for while, we've been silently converting it to `:simple` without warning the user. Do you think it would make sense to begin now to regard `:advanced` as an invalid humanization strategy - and make sure users know that this may break their installation (if, somehow, they've still got the humanization strategy set to `:advanced`)? I'm not sure what our policy is on deprecating features.

Reply:
Isn't the behavior in this PR treating invalid values as `:simple` as well, and just logging a warning? I'm just suggesting we should do the same for `:advanced`. So still coerce it to `:simple` like all other invalid values, and also log a warning that it's invalid. Shouldn't break anyone's installation.

Or am I misunderstanding the behavior here?

Reply:
Sorry, should have said "warning"